devtools/maven/src/main/java/io/quarkus/maven/CodeGenMojo.java

```diff
@@ -102,9 +102,8 @@ void doExecute(Path sourcesDir,
                     .setTargetDirectory(buildDir.toPath())
                     .build().bootstrap();
 
-            Path generatedSourcesDir = test
-                    ? buildDir.toPath().resolve("generated-test-sources")
-                    : buildDir.toPath().resolve("generated-sources");
+            Path generatedSourcesDir = buildDir.toPath().resolve(test ? "generated-test-sources" : "generated-sources")
+                    .resolve("quarkus");
```

Comment:
I think we may need both in case of testing.

Reply:
Or pass both to sourceRegistrar.accept(generatedSourcesDir) and keep the test dir CodeGenerator.init.

Reply:
Test execution comes from a separate goal. So there is one goal for the "main" part and one for the test part (if there is something to generate for tests at all), each one (potentially) adding a different directory.
Or do you mean the dev goal should also trigger prepare for the test part?

Reply:
Ah, right, it's invoked for each goal.